synapse/tools/aha/mirror.py

```diff
@@ -0,0 +1,195 @@
+                    outp.printf(f'Service at {opts.url} is not an Aha server')
+                    return 1
+
+                virtual_services, member_servers = {}, {}
+                async for svc in prox.getAhaSvcs():
+                    name = svc.get('name', '')
+                    svcinfo = svc.get('svcinfo', {})
+                    urlinfo = svcinfo.get('urlinfo', {})
+                    hostname = urlinfo.get('hostname', '')
+
+                    if name != hostname:
+                        virtual_services[name] = svc
+                    else:
+                        member_servers[name] = svc
+
+                mirror_groups = {}
+                for vname, vsvc in virtual_services.items():
+                    vsvc_info = vsvc.get('svcinfo', {})
+                    vsvc_iden = vsvc_info.get('iden')
+                    vsvc_leader = vsvc_info.get('leader')
+                    vsvc_hostname = vsvc_info.get('urlinfo', {}).get('hostname', '')
+
+                    if not vsvc_iden or not vsvc_hostname or not vsvc_leader:
+                        continue
+
+                    primary_member = member_servers.get(vsvc_hostname)
+                    if not primary_member:
+                        continue
+
+                    members = [primary_member] + [
+                        msvc for mname, msvc in member_servers.items()
+                        if mname != vsvc_hostname and
+                        msvc.get('svcinfo', {}).get('iden') == vsvc_iden and
+                        msvc.get('svcinfo', {}).get('leader') == vsvc_leader
+                    ]
+
+                    if len(members) > 1:
+                        mirror_groups[vname] = members
+
+                outp.printf('Service Mirror Groups:')
+                for vname, members in mirror_groups.items():
+                    iden = members[0].get('svcinfo', {}).get('iden')
+
+                    cell_infos = await get_cell_infos(prox, iden, members, opts.timeout)
+                    group_status = build_status_list(members, cell_infos)
+                    output_status(outp, vname, group_status)
+
+                    if check_sync_status(group_status):
+                        outp.printf('Group Status: In Sync')
+                    else:
+                        outp.printf(f'Group Status: Out of Sync; Waiting {opts.timeout} seconds for sync...')
+                        leader_nexs = None
+                        for status in group_status:
+                            if status['role'] == 'leader' and isinstance(status['nexs_indx'], int):
+                                leader_nexs = status['nexs_indx']
+                                break
+
+                        if leader_nexs is not None:
+                            start_time = s_common.now()
+
+                            while True:
+                                now = s_common.now()
+                                remaining_ms = (opts.timeout * 1000) - (now - start_time)
```

Comment:
If we pass in a negative timeout value, i think we would throw the timeout exception right away? We could raise a BadArg value instead.

Reply:
Done..!